### PR TITLE
[Fix] keras.ops.flip fails for eager tensors when axis is a list of axes

### PR DESCRIPTION
**Pull request by ChiragSW**
Fixes issue: #22539

### CODE REVIEW

**Comment by ChiragSW** [comment]
Please review, the issue has been resolved

**Comment by keerthanakadiri** [comment]
Hi @ChiragSW, Can you please provide more information and how this PR would solve the issue.

**Comment by ChiragSW** [comment]
## changes
Backend update: Refactored the flip call in the TF backend to pass the axis directly to tf.reverse.

Added new test cases in ```keras/ops/math_test.py``` that specifically test flipping with axis=(-1,) and axis=(0, 2) to prevent future regressions.

**Comment by ChiragSW** [comment]
@keerthanakadiri any updates needed?

File: keras/src/backend/tensorflow/numpy.py [hunk 1]
```diff
@@ -1590,7 +1590,10 @@ def flip(x, axis=None):
     x = convert_to_tensor(x)
     if axis is None:
         return tf.reverse(x, tf.range(tf.rank(x)))
-    return tf.reverse(x, [axis])
+    axis = to_tuple_or_list(axis)
+    axis = tf.convert_to_tensor(axis, dtype=tf.int32)
+    axis = tf.where(axis < 0, axis + tf.rank(x), axis)
```
Comment (hertschuh) [inline keras/src/backend/tensorflow/numpy.py]: Let's handle the axis statically as they're supposed to be ints, not tensors.
Also, `canonicalize_axis` does some validation:

```python
rank = x.shape.rank

if axis is None:
    return tf.reverse(x, tf.range(rank))
else:
    axis = to_tuple_or_list(axis)
    axis = [canonicalize_axis(a, rank) for a in axis]
    return tf.reverse(x, axis)
```

**Review by hertschuh — CHANGES_REQUESTED** [review]
Thanks for looking into this issue!
